src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/PreviousQualificationsEntityMapper.kt

```diff
@@ -126,31 +130,93 @@ abstract class PreviousQualificationsEntityMapper {
       )
     }
 
-  fun toEducationLevel(educationLevel: EducationLevelDomain): EducationLevelEntity =
-    when (educationLevel) {
-      EducationLevelDomain.NOT_SURE -> EducationLevelEntity.NOT_SURE
-      EducationLevelDomain.PRIMARY_SCHOOL -> EducationLevelEntity.PRIMARY_SCHOOL
-      EducationLevelDomain.SECONDARY_SCHOOL_LEFT_BEFORE_TAKING_EXAMS -> EducationLevelEntity.SECONDARY_SCHOOL_LEFT_BEFORE_TAKING_EXAMS
-      EducationLevelDomain.SECONDARY_SCHOOL_TOOK_EXAMS -> EducationLevelEntity.SECONDARY_SCHOOL_TOOK_EXAMS
-      EducationLevelDomain.FURTHER_EDUCATION_COLLEGE -> EducationLevelEntity.FURTHER_EDUCATION_COLLEGE
-      EducationLevelDomain.UNDERGRADUATE_DEGREE_AT_UNIVERSITY -> EducationLevelEntity.UNDERGRADUATE_DEGREE_AT_UNIVERSITY
-      EducationLevelDomain.POSTGRADUATE_DEGREE_AT_UNIVERSITY -> EducationLevelEntity.POSTGRADUATE_DEGREE_AT_UNIVERSITY
+  private fun qualificationDataHasChanged(qualificationEntityToUpdate: QualificationEntity, updateQualificationDto: UpdateQualificationDto): Boolean =
+    with(qualificationEntityToUpdate) {
+      !(subject == updateQualificationDto.subject && grade == updateQualificationDto.grade && toQualificationLevel(level!!) == updateQualificationDto.level)
     }
 }
 
-@Mapper
-interface QualificationEntityMapper {
-  @ExcludeJpaManagedFields
-  @GenerateNewReference
-  @ExcludeParentEntity
-  fun fromDomainToEntity(domain: UpdateOrCreateQualificationDto): QualificationEntity
-
-  @ExcludeJpaManagedFields
-  @ExcludeReferenceField
-  @ExcludeParentEntity
-  fun updateEntityFromDomain(@MappingTarget entity: QualificationEntity, domain: UpdateOrCreateQualificationDto)
-
-  @Mapping(target = "lastUpdatedBy", source = "updatedBy")
-  @Mapping(target = "lastUpdatedAt", source = "updatedAt")
-  fun fromEntityToDomain(persistedEntity: QualificationEntity): Qualification
+@Component
```

Comment:
No longer a mapstruct mapper. It's now longer in terms of lines of code, but arguably is better/more readable/more maintainable 🤷‍♂️

Reply:
and less warning in the build :-)